qiskit_dynamics/signals/signals.py

```diff
@@ -675,12 +642,12 @@ def __init__(
             name: name of the signal.
         """
 
-        samples = Array(samples)
+        samples = unp.asarray(samples)
         if carrier_freq is None:
-            carrier_freq = np.zeros(samples.shape[-1], dtype=float)
+            carrier_freq = unp.zeros(samples.shape[-1], dtype=float)
```

Comment:
This can stay as `np` as `samples.shape[-1]` will always be a standard python integer.
```suggestion
            carrier_freq = np.zeros(samples.shape[-1], dtype=float)
```